### PR TITLE
Feature/flag include year

### PR DESCRIPTION
**Pull request by shaun-nx**
### Proposed changes
This change introduces a new flag to the ingress controller `-include-year` which will add the year to the log header when set. The default value of this flag is `false`

### Deprecation notice
This flag is planned to be remove from the ingress controller by release 2.7 at which point all logs from the ingress controller will display the year by default.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by jjngx — APPROVED** [review]
👍🏻 great job!

File: deployments/deployment/nginx-plus-ingress.yaml [hunk 1]
```diff
@@ -65,6 +65,7 @@ spec:
           - -nginx-plus
           - -nginx-configmaps=$(POD_NAMESPACE)/nginx-config
           - -default-server-tls-secret=$(POD_NAMESPACE)/default-server-secret
+         #- -include-year
```
Comment (vepatel) [inline deployments/deployment/nginx-plus-ingress.yaml:68]: can specify the same in `deployments/daemon-set/nginx-plus-ingress.yaml` as well

Comment (shaun-nx) [reply deployments/deployment/nginx-plus-ingress.yaml:68]: Good catch. I will also add it to the OSS version of the deployments

File: deployments/helm-chart/values.yaml [hunk 2]
```diff
@@ -178,6 +178,9 @@ controller:
   ## Enable OIDC policies.
   enableOIDC: false
 
+  ## Include year in log header. This parameter will be removed in release 2.7 and the year will be included by default.
+  includeYear: false
```
Comment (vepatel) [inline deployments/helm-chart/values.yaml:182]: a corresponding entry in https://github.com/nginxinc/kubernetes-ingress/blob/feature/flag-include-year/docs/content/installation/installation-with-helm.md  as `controller.includeYear`

Comment (shaun-nx) [reply deployments/helm-chart/values.yaml:182]: Not sure I understand this. None of the values in this file start with `controller.` why would the `includeYear` value need to start with that?

Comment (vepatel) [reply deployments/helm-chart/values.yaml:182]: it refers to https://github.com/nginxinc/kubernetes-ingress/blob/feature/flag-include-year/deployments/helm-chart/values.yaml#L1 when you're installing using ` helm install nginx-ingress nginx-stable/nginx-ingress --set controller.kind=daemonset --set controller.includeYear=true ......`

**Review by vepatel — APPROVED** [review]

**Pull request merged**